pos_order_line_no_unlink/models/pos_order_line.py

```diff
@@ -0,0 +1,20 @@
+# Copyright (C) 2021 - Today: GRAP (http://www.grap.coop)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, models
+from odoo.exceptions import UserError
+
+
+class PosOrderLine(models.Model):
+    _inherit = "pos.order.line"
+
+    @api.multi
+    def unlink(self):
+        if self.filtered(
+            lambda x: x.order_id.state not in ["draft", "cancel"]
```

Comment:
I'll only allow it for draft.

Reply:
Well, it is for consistency reason with the current design.
https://github.com/odoo/odoo/blob/12.0/addons/point_of_sale/models/pos_order.py#L700

in fact, 'cancel' orders are not possible in V12. it is kept only for legacy reason, I guess.
see : https://github.com/odoo/odoo/blob/12.0/addons/point_of_sale/models/pos_order.py#L785